src/server/utils.js

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-underscore-dangle */
+import R from 'ramda';
+import S from 'sanctuary';
+import { simpleClique } from 'cliquer';
+
+// Make a map of lenses from a list of strings
+export const makeLenses = simpleClique(R.lensProp);
+
+// Version of R.view that returns `Just` the value or `Nothing`
+export const viewSafe = S.encase2(R.view);
+
+// Takes a lens and a list, acts like `R.pluck` using the lens
+// as a property accessor and returns `Just` an array of the results
+export const deepPluck = S.curry2((lens, xs) =>
+  S.pipe([S.map(viewSafe(lens)), S.justs])(xs));
+
+const prepend = R.ifElse(
+  R.test(/^measure/),
+  R.concat('$measures.'),
+  R.concat('$')
+);
+
+const aggregationOperator = R.cond([
+  [R.equals('count'), R.always({ $sum: 1 })],
+  [R.equals('month'), R.always({ $first: { $month: prepend('date') } })],
+  [R.T, key => ({ $first: prepend(key) })],
+]);
+
+const formId = R.cond([
+  [R.equals('count'), R.always('')],
+  [
+    R.equals('month'),
+    R.always({ $substr: [{ $month: prepend('date') }, 0, -1] }),
+  ],
+  [R.T, key => ({ $substr: [prepend(key), 0, -1] })],
+]);
+
+const generateId = S.pipe([
+  R.map(formId),
+  R.intersperse('|'),
+  R.objOf('$concat'),
+]);
+
+const L = makeLenses([
+  'name',
+  'operation',
+  'resolveInfo',
+  'selections',
+  'selectionSet',
+  'value',
+  0,
+]);
+
+export const lenses = L;
+
+export const getQueryKeys = S.pipe([
+  // this composed lens could be exported used with `set` or `over` as well
+  viewSafe(
```

Comment:
Personally I find this much less readable than the original implementation. The original implementation is pretty much pure JS, and should be accessible to anyone who knows what the form of the object is (which can be found pretty easily through debugging). This adds a level of indirection (now you have to understand lenses, sanctuary.js, and ramda to understand this line) without significant gain -- we get perhaps better error messages, but I'd argue that we should be validating the input object instead (it would, again, be clearer).